protocols/kad/src/behaviour.rs

```diff
@@ -680,37 +685,50 @@ where
     ///
     /// The result of this operation is delivered in a
     /// [`KademliaEvent::OutboundQueryCompleted{QueryResult::GetRecord}`].
-    pub fn get_record(&mut self, key: record::Key, quorum: Quorum) -> QueryId {
-        let quorum = quorum.eval(self.queries.config().replication_factor);
-        let mut records = Vec::with_capacity(quorum.get());
-
-        if let Some(record) = self.store.get(&key) {
+    pub fn get_record(&mut self, key: record::Key) -> QueryId {
+        let record = if let Some(record) = self.store.get(&key) {
             if record.is_expired(Instant::now()) {
-                self.store.remove(&key)
+                self.store.remove(&key);
+                None
             } else {
-                records.push(PeerRecord {
+                Some(PeerRecord {
                     peer: None,
                     record: record.into_owned(),
-                });
+                })
             }
-        }
+        } else {
+            None
+        };
 
-        let done = records.len() >= quorum.get();
         let target = kbucket::Key::new(key.clone());
         let info = QueryInfo::GetRecord {
             key,
-            records,
-            quorum,
+            count: 1,
+            record_to_cache: record.as_ref().map(|r| r.record.clone()),
             cache_candidates: BTreeMap::new(),
         };
         let peers = self.kbuckets.closest_keys(&target);
         let inner = QueryInner::new(info);
         let id = self.queries.add_iter_closest(target.clone(), peers, inner); // (*)
```

Comment:
```suggestion
        let id = self.queries.add_iter_closest(target.clone(), peers, inner);
```

Is this reference this relevant?

Reply:
removed, not sure what this was from